packages/workflow/src/records/handler/eventNotificationHandler.ts

```diff
@@ -0,0 +1,107 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+import * as Hapi from '@hapi/hapi'
+import {
+  addExtensionsToTask,
+  Bundle,
+  getTaskFromSavedBundle,
+  KnownExtensionType,
+  resourceToSavedBundleEntry,
+  SavedTask,
+  StringExtensionType,
+  ValidRecord
+} from '@opencrvs/commons/types'
+import { getToken } from '@workflow/utils/auth-utils'
+import { indexBundle } from '@workflow/records/search'
+import { sendBundleToHearth, toSavedBundle } from '@workflow/records/fhir'
+import {
+  getLocationOrOfficeById,
+  getLoggedInPractitionerResource
+} from '@workflow/features/user/utils'
+import { internal } from '@hapi/boom'
+
+export async function eventNotificationHandler(
+  request: Hapi.Request,
+  h: Hapi.ResponseToolkit
+) {
+  const payload = request.payload as Bundle
+  const token = getToken(request)
+
+  const responseBundle = await sendBundleToHearth(payload)
+  const savedBundle = toSavedBundle(payload, responseBundle) as ValidRecord
+
+  const task = getTaskFromSavedBundle(savedBundle) as SavedTask
+  const systemInfo = await getLoggedInPractitionerResource(token)
+
+  const savedTaskWithRegLastUser = addExtensionsToTask(task, [
+    {
+      url: 'http://opencrvs.org/specs/extension/regLastUser',
+      valueReference: {
+        reference: `Practitioner/${systemInfo.id}`
+      }
+    }
+  ]) as SavedTask
```

Comment:
Is there a way to get rid of the type casts? I don't think these should be necessary 🤔

Reply:
Removed